src/contexts/addresses.tsx

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressInfo } from 'alephium-js/dist/api/api-explorer'
+import addressToGroup from 'alephium-js/dist/lib/address'
+import { TOTAL_NUMBER_OF_GROUPS } from 'alephium-js/dist/lib/constants'
+import { deriveNewAddressData } from 'alephium-js/dist/lib/wallet'
+import { merge } from 'lodash'
+import { createContext, FC, useCallback, useContext, useEffect, useState } from 'react'
+import { PartialDeep } from 'type-fest'
+
+import {
+  AddressSettings,
+  loadStoredAddressesIndexAndSettingsOfAccount,
+  storeAddressIndexAndSettingsOfAccount
+} from '../utils/addresses'
+import { useGlobalContext } from './global'
+
+export interface AddressesContextProps {
+  saveNewAddress: (address: AddressState) => void
+  addressesState: Map<string, AddressState>
```

Comment:
OK cool! Something else I want to point out related to this: It'd be nice if we start referring to the address hash as `AddressHash` instead of `address`. It'd make sense to use the variable name `address` when referring to objects containing many info about the address.

Reply:
Fully agreed 💯